api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3589,23 +3590,62 @@ public void setObjectRetention(
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    if (config == null) {
+
+    this.setObjectRetention(
+        SetObjectRetentionArgs.builder()
+            .bucket(bucketName)
+            .object(objectName)
+            .versionId(versionId)
+            .config(config)
+            .bypassGovernanceRetention(bypassGovernanceRetention)
```

Comment:
As per the conversion, it seems this method is not required. If required, my recommendation is to have method name as `overrideGovernance()`

Reply:
@balamurugana I looked at the older API spec https://docs.aws.amazon.com/AmazonS3/latest/API/archive-RESTObjectPUTRetention.html

This header is needed.

Reply:
No problem. I thought it was policy decision when you were talking about removing the flag.